### ISSUE TITLE
New Resource: aws_athena_workgroup

### ISSUE THREAD

**Issue by zerfahp** [terraform-providers/terraform-provider-aws#7650]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS released Athena Workgroups earlier this week. It would be very useful to have the ability to create these from Terraform.

"You can now use Amazon Athena Workgroups - A new resource type that can be used to separate query execution and query history between Users, Teams, or Applications running under the same AWS account. Because Workgroups act as resources, you can use resource-based policies to control access to a Workgroup. For example, if you have two teams or different applications using Athena, you can limit access by assigning them to different Workgroups. You can also separate ad hoc usage from scheduled reports by assigning them to different Workgroups. Queries running in one Workgroup are not visible to users or applications running in a different Workgroup."

### New or Affected Resource(s)

* aws_athena_workgroup

### Potential Terraform Configuration

```hcl
resource "aws_athena_workgroup" "hr_workgroup" {
    name                                = "HR Workgroup"
    description                         = "Athena Workgroup for HR"
    bytes_scanned_cutoff_per_query      = "${var.cut_off_bytes}"
    enforce_work_group_configuration    = "true"
    publish_cloudwatch_metrics_enabled  = "true"
    output_location                     = "${aws_s3_bucket.hr_bucket.arn}"
    encryption_option                   = "SSE_KMS"
    kms_key                             = "${aws_kms_key.hr_kms_key.arn}"    
}
```

### References

* https://aws.amazon.com/about-aws/whats-new/2019/02/athena_workgroups/

**Comment by zerfahp** [comment]
Is there any progress happening on this? It is currently not possible to provision these through cloudformation so Terraform could get ahead of the game here.

**Comment by bflad** [comment]
There is an open pull request available here: https://github.com/terraform-providers/terraform-provider-aws/pull/7995

**Comment by Rupesh2010** [comment]
looks like it is still not supported.

**Provider doesn't support resource: aws_athena_workgroup**

**Comment by bflad** [comment]
The new `aws_athena_workgroup` resource has been merged and will release with version 2.19.0 of the Terraform AWS Provider, likely later today. Shout out to @marcotesch who helped make this possible!

**Comment by zerfahp** [comment]
Thank you so much guys. Very much appreciated.

**Comment by bflad** [comment]
This has been released in [version 2.19.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2190-july-11-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

For further feature requests or bug reports with this functionality, please create a [new GitHub issue](https://github.com/terraform-providers/terraform-provider-aws/issues/new/choose) following the template for triage. Thanks!
